worlds/alttp/__init__.py

```diff
@@ -505,29 +505,29 @@ def collect_item(self, state: CollectionState, item: Item, remove=False):
     def pre_fill(self):
         from Fill import fill_restrictive, FillError
         attempts = 5
-        world = self.multiworld
-        player = self.player
-        all_state = world.get_all_state(use_cache=True)
+        all_state = self.multiworld.get_all_state(use_cache=False)
         crystals = [self.create_item(name) for name in ['Red Pendant', 'Blue Pendant', 'Green Pendant', 'Crystal 1', 'Crystal 2', 'Crystal 3', 'Crystal 4', 'Crystal 7', 'Crystal 5', 'Crystal 6']]
-        crystal_locations = [world.get_location('Turtle Rock - Prize', player),
-                             world.get_location('Eastern Palace - Prize', player),
-                             world.get_location('Desert Palace - Prize', player),
-                             world.get_location('Tower of Hera - Prize', player),
-                             world.get_location('Palace of Darkness - Prize', player),
-                             world.get_location('Thieves\' Town - Prize', player),
-                             world.get_location('Skull Woods - Prize', player),
-                             world.get_location('Swamp Palace - Prize', player),
-                             world.get_location('Ice Palace - Prize', player),
-                             world.get_location('Misery Mire - Prize', player)]
+        for crystal in crystals:
+            all_state.remove(crystal)
+        crystal_locations = [self.multiworld.get_location('Turtle Rock - Prize', self.player),
+                             self.multiworld.get_location('Eastern Palace - Prize', self.player),
+                             self.multiworld.get_location('Desert Palace - Prize', self.player),
+                             self.multiworld.get_location('Tower of Hera - Prize', self.player),
+                             self.multiworld.get_location('Palace of Darkness - Prize', self.player),
+                             self.multiworld.get_location('Thieves\' Town - Prize', self.player),
+                             self.multiworld.get_location('Skull Woods - Prize', self.player),
+                             self.multiworld.get_location('Swamp Palace - Prize', self.player),
+                             self.multiworld.get_location('Ice Palace - Prize', self.player),
+                             self.multiworld.get_location('Misery Mire - Prize', self.player)]
```

Comment:
```suggestion
        crystal_locations = [self.multiworld.get_location('Turtle Rock - Prize'),
                             self.get_location('Eastern Palace - Prize'),
                             self.get_location('Desert Palace - Prize'),
                             self.get_location('Tower of Hera - Prize'),
                             self.get_location('Palace of Darkness - Prize'),
                             self.get_location('Thieves\' Town - Prize'),
                             self.get_location('Skull Woods - Prize'),
                             self.get_location('Swamp Palace - Prize'),
                             self.get_location('Ice Palace - Prize'),
                             self.get_location('Misery Mire - Prize')]
```
Might as  well clean this up while here